qiskit/circuit/quantumcircuit.py

```diff
@@ -2564,9 +2564,26 @@ def add_calibration(self, gate, qubits, schedule, params=None):
             Exception: if the gate is of type string and params is None.
         """
         if isinstance(gate, Gate):
-            self._calibrations[gate.name][(tuple(qubits), tuple(gate.params))] = schedule
-        else:
-            self._calibrations[gate][(tuple(qubits), tuple(params or []))] = schedule
+            params = gate.params
+            gate = gate.name
+        self._calibrations[gate][(tuple(qubits), tuple(params or []))] = schedule
+        if params is not None:
+            self.add_calibration_params(gate, qubits, params)
+
+    def add_calibration_params(self, name, qubits, params):
```

Comment:
I think instead of tracking calibration params, we might have to try changing `assign_parameters` to just look through all the schedule calibrations for that parameter